src/content/dataSubmissions/QualityControl.tsx

```diff
@@ -0,0 +1,212 @@
+    context: { clientName: "backend" },
+    fetchPolicy: "no-cache",
+  });
+
+  useEffect(() => {
+    if (!submissionId || !isSubmissionOwner) {
+      return;
+    }
+    (async () => {
+      try {
+        setLoading(true);
+        const { data: d, error } = await listLogs();
+        if (error || !d?.listLogs) {
+          throw new Error("Unexpected network error");
+        }
+
+        setFiles(d.listLogs.logFiles);
+      } catch (err) {
+        setChangesAlert({
+          message: "Unable to retrieve logs",
+          severity: "error",
+        });
+        if (alertTimeoutRef.current) {
+          clearTimeout(alertTimeoutRef.current);
+        }
+        alertTimeoutRef.current = setTimeout(
+          () => setChangesAlert(null),
+          10000
+        );
+      } finally {
+        setLoading(false);
+      }
+    })();
+  }, [submissionId, submitterID, user]);
+
+  const downloadFiles = async (uploadType: UploadType) => {
+    const zip = new JSZip();
+    const filesToDownload = files.filter(
+      (file) => file.uploadType === uploadType
+    );
+
+    await Promise.all(
+      filesToDownload.map(async (file) => {
+        const response = await fetch(file.downloadUrl);
+        const blob = await response.blob();
+        zip.file(file.fileName, blob);
+      })
+    );
+
+    zip.generateAsync({ type: "blob" }).then((blob) => {
+      saveAs(blob, `${uploadType}_logs.zip`);
+    });
+  };
+
+  return (
+    <>
+      <GenericAlert
+        open={!!changesAlert}
+        severity={changesAlert?.severity}
+        key="submission-quality-control-changes-alert"
+      >
+        <span>{changesAlert?.message}</span>
+      </GenericAlert>
+      <Stack
+        direction="column"
+        alignItems="center"
+        paddingTop="66px"
+        spacing={5.5}
+      >
+        <Stack direction="column" spacing={3}>
+          <Stack
+            direction="row"
+            spacing={2.625}
+            justifyContent="end"
+            alignItems="center"
+          >
+            <StyledDownloadLabel
+              variant="body1"
+              textAlign="end"
+              sx={{ color: !fileLogs?.length ? "#AAA" : "083A50" }}
+            >
+              Files Validation Results
+            </StyledDownloadLabel>
+            <StyledDownloadButton
+              onClick={() => downloadFiles("file")}
+              loading={loading}
+              disabled={!fileLogs?.length}
+            >
+              Download
+            </StyledDownloadButton>
+          </Stack>
+          <Stack
+            direction="row"
+            spacing={2.625}
+            justifyContent="end"
+            alignItems="center"
+          >
+            <StyledDownloadLabel
+              variant="body1"
+              textAlign="end"
+              sx={{ color: !metadataLogs?.length ? "#AAA" : "083A50" }}
+            >
+              Metadata Validation Results
+            </StyledDownloadLabel>
+            <StyledDownloadButton
```

Comment:
For these two download buttons, one change I would suggest is to throw it in the loading state while downloading/zipping the files. If you have slow internet, it may look like nothing is happening. Clicking fast also downloads multiple times, which may not be desirable. 

This can wait until [CRDCDH-554](https://tracker.nci.nih.gov/browse/CRDCDH-554).

Reply:
Added state to set button to loading while downloading. As well as a missing try/catch